Align AliasInfo's operator<< with FunctionSchema

houseroad opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23422 [WIP] Add CI job to check function schemas' backward compatiblity
* #23409 Add isBackwardCompatibleWith for Argument and FunctionSchema
* #23208 expose parse_schema and __eq__ function to python and add round trip tests
* #23207 Expose an API to iterate all the registered operators
* **#23206 Align AliasInfo's operator<< with FunctionSchema**

old (a)
new (a! -> b)

Differential Revision: [D16433909](https://our.internmc.facebook.com/intern/diff/D16433909/)

houseroad committed:
Align AliasInfo's operator<< with FunctionSchema

old (a)
new (a! -> b)

Differential Revision: [D16433909](https://our.internmc.facebook.com/intern/diff/D16433909/)

houseroad committed:
Update on "Align AliasInfo's operator<< with FunctionSchema"


old (a)
new (a! -> b)

Differential Revision: [D16433909](https://our.internmc.facebook.com/intern/diff/D16433909/)

houseroad committed:
Update on "Align AliasInfo's operator<< with FunctionSchema"


old (a)
new (a! -> b)

Differential Revision: [D16433909](https://our.internmc.facebook.com/intern/diff/D16433909/)

houseroad committed:
Update on "Align AliasInfo's operator<< with FunctionSchema"


old (a)
new (a! -> b)

Differential Revision: [D16433909](https://our.internmc.facebook.com/intern/diff/D16433909/)

dzhulgakov commented on aten/src/ATen/core/alias_info.h:
```diff
@@ -98,11 +98,22 @@ inline std::ostream& operator<<(std::ostream& out, const AliasInfo& aliasInfo) {
   }
-  out << ")";
-
-  if (!aliasInfo.containedTypes().empty()) {
```
not sure whether removing it makes debugging harder or not. We can also put a separate method to print to string, but @suo is best to decide

houseroad committed:
Update on "Align AliasInfo's operator<< with FunctionSchema"


old (a)
new (a! -> b)

Differential Revision: [D16433909](https://our.internmc.facebook.com/intern/diff/D16433909/)

houseroad committed:
Update on "Align AliasInfo's operator<< with FunctionSchema"


old (a)
new (a! -> b)

Differential Revision: [D16433909](https://our.internmc.facebook.com/intern/diff/D16433909/)

dzhulgakov reviewed (approved):
I'm stamping it just to unblock

houseroad committed:
Update on "Align AliasInfo's operator<< with FunctionSchema"


old (a)
new (a! -> b)

Differential Revision: [D16433909](https://our.internmc.facebook.com/intern/diff/D16433909/)
